pkg/controller/operators/operator_controller.go

```diff
@@ -0,0 +1,259 @@
+	apiregistrationv1 "k8s.io/kube-aggregator/pkg/apis/apiregistration/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	"github.com/operator-framework/operator-lifecycle-manager/pkg/controller/operators/decorators"
+)
+
+var (
+	localSchemeBuilder = runtime.NewSchemeBuilder(
+		kscheme.AddToScheme,
+		apiextensionsv1.AddToScheme,
+		apiregistrationv1.AddToScheme,
+		operatorsv1alpha1.AddToScheme,
+		operatorsv1.AddToScheme,
+		operatorsv2alpha1.AddToScheme,
+	)
+
+	// AddToScheme adds all types necessary for the controller to operate.
+	AddToScheme = localSchemeBuilder.AddToScheme
+)
+
+// OperatorReconciler reconciles a Operator object.
+type OperatorReconciler struct {
+	client.Client
+
+	log     logr.Logger
+	mu      sync.RWMutex
+	factory decorators.OperatorFactory
+
+	// operators contains the names of Operators the OperatorReconciler has observed exist.
+	operators map[types.NamespacedName]struct{}
+}
+
+// +kubebuilder:rbac:groups=operators.coreos.com,resources=operators,verbs=create;update;patch;delete
+// +kubebuilder:rbac:groups=operators.coreos.com,resources=operators/status,verbs=update;patch
+// +kubebuilder:rbac:groups=*,resources=*,verbs=get;list;watch
+
+// SetupWithManager adds the operator reconciler to the given controller manager.
+func (r *OperatorReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	// Trigger operator events from the events of their compoenents.
+	enqueueOperator := &handler.EnqueueRequestsFromMapFunc{
+		ToRequests: handler.ToRequestsFunc(r.mapComponentRequests),
+	}
+
+	// Note: If we want to support resources composed of custom resources, we need to figure out how
+	// to dynamically add resource types to watch.
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&operatorsv2alpha1.Operator{}).
+		Watches(&source.Kind{Type: &appsv1.Deployment{}}, enqueueOperator).
+		Watches(&source.Kind{Type: &corev1.Namespace{}}, enqueueOperator).
+		Watches(&source.Kind{Type: &corev1.ServiceAccount{}}, enqueueOperator).
+		Watches(&source.Kind{Type: &corev1.Secret{}}, enqueueOperator).
+		Watches(&source.Kind{Type: &rbacv1.Role{}}, enqueueOperator).
+		Watches(&source.Kind{Type: &rbacv1.RoleBinding{}}, enqueueOperator).
+		Watches(&source.Kind{Type: &rbacv1.ClusterRole{}}, enqueueOperator).
+		Watches(&source.Kind{Type: &rbacv1.ClusterRoleBinding{}}, enqueueOperator).
+		Watches(&source.Kind{Type: &apiextensionsv1.CustomResourceDefinition{}}, enqueueOperator).
+		Watches(&source.Kind{Type: &apiregistrationv1.APIService{}}, enqueueOperator).
+		Watches(&source.Kind{Type: &operatorsv1alpha1.Subscription{}}, enqueueOperator).
+		Watches(&source.Kind{Type: &operatorsv1alpha1.InstallPlan{}}, enqueueOperator).
+		Watches(&source.Kind{Type: &operatorsv1alpha1.ClusterServiceVersion{}}, enqueueOperator).
```

Comment:
This list needs to be kept in sync with the supported bundle types - i.e. configmap and secret support will merge soon.

Reply:
true -- not sure how we want to handle it. This is a stopgap for generic "watch all" behavior.